src/session_pool/session_pool.js

```diff
@@ -168,26 +169,58 @@ export class SessionPool extends EventEmitter {
         events.on(ACTOR_EVENT_NAMES_EX.PERSIST_STATE, this._listener);
     }
 
+    /**
+     * Explicitly creates and adds new session to the session pool.
+     * @param {SessionOptions} [options] - The configuration options for the session being added to the session pool.
+     */
+    async addSession(options = {}) {
+        const { id } = options;
+        if (id) {
+            const sessionExists = this.sessionMap.has(id);
+            if (sessionExists) {
+                throw new Error(`Cannot add session with id '${id}' as it already exists in the pool`);
+            }
+        }
+
+        if (!this._hasSpaceForSession()) {
+            this._removeRetiredSessions();
+        }
+
+        const newSession = await this.createSessionFunction(this, { sessionOptions: options });
```

Comment:
I assume this could be considered resolved? Works both with `Session` and `SessionOptions` but would throw if a user would try to add new session with existing `id`.